src/js/extensions/toolbar.js

```diff
@@ -616,35 +616,72 @@
                 }
             }
 
-            var windowWidth = this.window.innerWidth,
-                middleBoundary = (boundary.left + boundary.right) / 2,
+            var containerWidth = this.window.innerWidth,
                 toolbarElement = this.getToolbarElement(),
                 toolbarHeight = toolbarElement.offsetHeight,
                 toolbarWidth = toolbarElement.offsetWidth,
                 halfOffsetWidth = toolbarWidth / 2,
                 buttonHeight = 50,
-                defaultLeft = this.diffLeft - halfOffsetWidth;
+                defaultLeft = this.diffLeft - halfOffsetWidth,
+                elementsContainer = this.getEditorOption('elementsContainer'),
+                elementsContainerAbsolute = ['absolute', 'fixed'].indexOf(window.getComputedStyle(elementsContainer).getPropertyValue('position')) > -1,
+                positions = {},
+                relativeBoundary = {},
+                middleBoundary, elementsContainerBoundary;
+
+            // If container element is absolute / fixed, recalculate boundaries to be relative to the container
+            if (elementsContainerAbsolute) {
+                elementsContainerBoundary = elementsContainer.getBoundingClientRect();
+                ['top', 'left'].forEach(function (key) {
+                    relativeBoundary[key] = boundary[key] - elementsContainerBoundary[key];
+                });
+
+                ['bottom', 'right'].forEach(function (key) {
+                    relativeBoundary[key] = elementsContainerBoundary[key] - boundary[key];
+                });
+
+                relativeBoundary.width = boundary.width;
+                relativeBoundary.height = boundary.height;
+                boundary = relativeBoundary;
+
+                containerWidth = elementsContainerBoundary.width;
+            }
+
+            middleBoundary = boundary.left + boundary.width / 2;
+            positions.top = boundary.top - toolbarHeight;
+
+            // If container element is absolute / fixed, adjust top position according to container scroll position
+            if (elementsContainerAbsolute) {
+                positions.top += elementsContainer.scrollTop;
+            // If container element isn't absolute / fixed, adjust top position according to window scroll position
+            } else {
+                positions.top += this.window.pageYOffset;
```

Comment:
Could the calculation of the adjustment of top be done within the `if(elementsContainerAboslute)` block above?

Something like:
```
var topAdjustment = this.window.pageYOffset;

 // If container element is absolute / fixed, recalculate boundaries to be relative to the container
 if (elementsContainerAbsolute) {
    // other existing logic...

    topAdjustment = elementsContainer.scrollTop;
}

middleBoundary = boundary.left + boundary.width / 2;
positions.top = (boundary.top - toolbarHeight) + topAdjustment;
```

That way we can have one less if-condition.